server/cloudformation/environment.go

```diff
@@ -0,0 +1,160 @@
+package cloudformation
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/remind101/empire"
+	"golang.org/x/net/context"
+)
+
+type EnvironmentResource struct {
+	empire *empire.Empire
+}
+
+type VariableError struct {
+	index int
+	err   string
+}
+
+func (v *VariableError) Error() string {
+	return fmt.Sprintf("invalid variable [%d]: %s", v.index, v.err)
+}
+
+func (p *EnvironmentResource) Provision(req Request) (id string, data interface{}, err error) {
+	ctx := context.Background()
+	user := NewUser()
+
+	var ok bool
+	switch req.RequestType {
+	case Create:
+		id, ok = req.ResourceProperties["AppId"].(string)
+		if !ok {
+			return "", nil, fmt.Errorf("missing parameter: AppId")
+		}
+	default:
+		id = req.PhysicalResourceId
+	}
+
+	app, err := p.empire.AppsFind(empire.AppsQuery{
+		ID: &id,
+	})
+	if err != nil {
+		return id, nil, err
+	}
+
+	if err := p.setEnvironment(ctx, user, app, req); err != nil {
+		return id, nil, err
+	}
+	return
+}
+
+func (p *EnvironmentResource) setEnvironment(ctx context.Context, user *empire.User, app *empire.App, req Request) error {
+	vars, err := varsFromRequest(req)
+	if err != nil {
+		return err
+	}
+
+	var action string
+	switch req.RequestType {
+	case Create:
+		action = "Setting"
+	case Update:
+		action = "Updating"
+	case Delete:
+		action = "Unsetting"
+	}
+
+	_, err = p.empire.Set(ctx, empire.SetOpts{
+		User:    user,
+		App:     app,
+		Vars:    vars,
+		Message: fmt.Sprintf("%s variables via Cloudformation", action),
+	})
+
+	return err
+}
+
+type Variable struct {
+	Key   string
+	Value *string
+}
+
+func parseVariable(index int, input interface{}) (*Variable, error) {
```

Comment:
We should just change the type of `ResourceProperties` and `OldResourceProperties` types to json.Message, then we can just unmarshal into a struct. I already did this in https://github.com/remind101/empire/pull/842, but can break it out into a smaller PR that we can merge first.